asab/zookeeper/container.py

```diff
@@ -94,9 +95,19 @@ async def _do_advertise(self, zoocontainer):
 				await zoocontainer.ZooKeeper.set_data(self.Node, self.Data)
 				return
 
-			self.Node = await zoocontainer.ZooKeeper.create(
-				self.Path,
-				data=self.Data,
-				sequential=True,
-				ephemeral=True
-			)
+			try:
+				self.Node = await zoocontainer.ZooKeeper.create(
+					self.Path,
+					data=self.Data,
+					sequential=True,
+					ephemeral=True
+				)
+			except aiozk.exc.NoNode:
+				missing_node = self.Path.rstrip(self.Path.split("/")[-1])
+				await zoocontainer.ZooKeeper.create(missing_node)
+				self.Node = await zoocontainer.ZooKeeper.create(
+					self.Path,
```

Comment:
... or `await zoocontainer.ZooKeeper.create(` could be wrapped in the inline function.

Reply:
@ateska I think the /run folder/node should be created at APIService constructor IMHO.

Reply:
yes, that's a correct point

Reply:
API Service methods are synchronous, while communication with Zookeeper is asynchronous. That's why there's ZooKeeperAdvertisement class, I guess, and that's why I tried to fix the issue quite "far" from where it originates.